src/Transition.js

```diff
@@ -322,18 +322,14 @@ class Transition extends React.Component {
     return this.nextCallback
   }
 
-  onTransitionEnd(node, timeout, handler) {
+  onTransitionEnd(node, timeout, handler, status) {
     this.setNextCallback(handler)
 
-    if (node) {
```

Comment:
@jquense [GSAP use case](https://github.com/mvasin/rtg-gsap-example/blob/master/src/App.js#L10-L22) does not oblige to provide a DOM node; it can animate a bunch of elements by their classes or ids (and not necessary this `node` element among them), so I removed this check.

Reply:
@jquense According to `git blame`, you are the one who wrote this line. If you have no idea what it stands for, let's remove it.

Reply:
I do have idea what it stands for, that's what I'm saying...it prevents edgecase bugs where nodes are detached or mounted/recycled. This isn't an unsure about old code concern, I'm saying that it's a needed check

Reply:
I believe you and am ready to put it back. But it would be a healthy practice to write a test that fails without this conditional and passes with it. Can you write one?

And I'll try to write a test for GSAP use case that fails with this conditional. :) Hopefully we'll find a middle ground.

Reply:
I added tests that ensure addEndListener is triggered even when `<Transition>`'s child renders nothing to the DOM. I think this behaviour is the most versatile; it works well when a child only orchestrates animations and does not render any DOM elements by itself.

It _does_ fail if the `if (node)` conditional is put back.

An interesting issue arose - have a look at [the failing test on Travis](https://travis-ci.org/reactjs/react-transition-group/builds/416226274). If you accept that `<Transition>` can have a child renders nothing to the DOM, I'll investigate it.

Reply:
looking good, tho I don't think i understand why the "renders no node" case is important here. It looks like the example you've provided does render nodes, e.g. the page associated with the route.

Reply:
I've used `() => null` component that does not render to DOM and imitates a GSAP component that does timeline animation by ids/classes of other components and does not render to DOM anything on its own.

When `<Transition>` contains `() => null` (named `wrapperDOMlessChild` in the tests), you can see in the failing test that when exiting starts, `<Transition>` is not in `entered` state as expected, but in `exited` instead.

Reply:
Can we split that that case into another PR, and keep this focused on adding the status to the endListener? I think we are getting a bit sidetracked on a thing that wasn't the original functional addition